js/classes/entities/midground/weapon/laser.js

```diff
@@ -0,0 +1,154 @@
+    static get INITIAL_SPEED() {
+        return 6;
+    }
+
+    /** The weight of the Laser. */
+    static get WEIGHT() {
+        return 0;
+    }
+
+    /** The amount of time the laser will stay alive after firing */
+    static get ALIVE_TIME() {
+        return 7;
+    }
+
+    /** The file path to the Laser's spritesheet. */
+    static get SPRITESHEET() {
+        return "./sprites/projectile_laser_yellow.png";
+    }
+
+    static get DAMAGE() {
+        return 7;
+    }
+
+    /** Called when the laser collides with a block. */
+    onBlockCollision(block) {
+        this.removeFromWorld = true;
+    }
+
+    /** Called when the laser collides with an enemy. */
+    onEnemyCollision(enemy) {
+        // do nothing
+    }
+
+    /** Called when the laser collides with the player. */
+    onPlayerCollision(player) {
+        if (!this.hasHit) {
+            player.takeDamage(Laser.DAMAGE);
+            this.hasHit = true;
+            this.removeFromWorld = true;
+        }
+    }
+
+    loadAnimations() {
+        this.animations["firing"] = [];
+
+        this.animations["firing"]["right"] = new Animator(
+            Laser.SPRITESHEET, 
+            new Vector(0, Laser.SIZE.y * 0), 
+            Laser.SIZE, 1, 0.5); // 0 degree angle
+
+        this.animations["firing"]["rightDown"] = new Animator(
+            Laser.SPRITESHEET, 
+            new Vector(0, Laser.SIZE.y * 1), 
+            Laser.SIZE, 1, 0.5); // 45 degree angle
+
+        this.animations["firing"]["down"] = new Animator(
+            Laser.SPRITESHEET, 
+            new Vector(0, Laser.SIZE.y * 2), 
+            Laser.SIZE, 1, 0.5); // 90 degree angle
+
+        this.animations["firing"]["leftDown"] = new Animator(
+            Laser.SPRITESHEET, 
+            new Vector(0, Laser.SIZE.y * 3), 
+            Laser.SIZE, 1, 0.5); // 135 degree angle
+
+        this.animations["firing"]["left"] = new Animator(
+            Laser.SPRITESHEET, 
+            new Vector(0, Laser.SIZE.y * 4), 
+            Laser.SIZE, 1, 0.5); // 180 degree angle
+
+        this.animations["firing"]["leftUp"] = new Animator(
+            Laser.SPRITESHEET, 
+            new Vector(0, Laser.SIZE.y * 5), 
+            Laser.SIZE, 1, 0.5); // 225 degree angle
+
+        this.animations["firing"]["up"] = new Animator(
+            Laser.SPRITESHEET,
+            new Vector(0, Laser.SIZE.y * 6), 
+            Laser.SIZE, 1, 0.5); // 270 degree angle
+
+        this.animations["firing"]["rightUp"] = new Animator(
+            Laser.SPRITESHEET, 
+            new Vector(0, Laser.SIZE.y * 7), 
+            Laser.SIZE, 1, 0.5); // 315 degree angle
+    }
+
+    update() {
+        this.base.update();
+
+        this.aliveTimer += GAME.clockTick;
+        if (this.aliveTimer >= Laser.ALIVE_TIME) {
+            this.removeFromWorld = true;
+        }
+
+        if (this.action == "firing" && !this.hasHit && GAME.gameTime % 0.1 < 0.01) {
```

Comment:
What are these magic numbers?

Reply:
I probably should have made constants for those. `GAME.gameTime % 0.1` is saying "every 1/10th of a second". `< 0.01` is giving some margin of error for  `GAME.gameTime % 0.1` to result in 0.

Reply:
I don't agree that you should make constants for these. Maybe a comment could be helpful, but constants for things like this seem like overkill to me.

Reply:
We can do it either way, it won't make much of a difference to me at this point however I still wonder if using local variables or constants if they are reused values would improve readability which may be worth it.

Reply:
Good point. I'll add some constants